lib/client.js

```diff
@@ -1769,7 +1769,7 @@ function writeH2 (client, session, request) {
 
   session.ref()
 
-  const shouldEndStream = method === 'GET' || method === 'HEAD'
+  const shouldEndStream = method === 'GET' || method === 'HEAD' || body === null
```

Comment:
```suggestion
  const shouldEndStream = method === 'GET' || method === 'HEAD' || body == null
```

To also handle `undefined`

Reply:
A test for that would be needed as well

Reply:
Here, if body is undefined, it will be null, so this change does not seem necessary.
https://github.com/nodejs/undici/blob/03f419f3ccee0c7779cf0dbc7b8d3a16e9eee0a2/lib/core/request.js#L95-L96

Reply:
Good catch, although I'll still add the test for both cases 👍